src/celldega/nbhd/__init__.py

```diff
@@ -300,4 +305,208 @@ def create_hextile(radius, path_landscape_files=None, img_height=100, img_width=
+
+    This function reads transcript data and transformation parameters from the given directories,
+    applies coordinate transformations if necessary, allots transcripts to hexagonal spatial tiles,
+    and calculates the percentage of unassigned transcripts per tile.
+
+    Tiles with a percentage of unassigned transcripts lower than the specified threshold are visualized.
+
+    Parameters:
+    -----------
+    gdf_hextile : geopandas.GeoDataFrame
+        A GeoDataFrame containing the hexagonal tiling of the spatial region with associated geometry.
+
+    path_data : str
+        Path to the directory containing transcriptomic data, including `transcripts.parquet`.
+
+    percentage_unassigned_threshold : float, optional (default=75)
+        The threshold (in percent) for including tiles in the visualization. Tiles with a higher
+        percentage of unassigned transcripts are excluded from the plot.
+
+    Returns:
+    --------
+    None
+        This function displays a matplotlib plot showing the percentage of unassigned transcripts
+        in each hexagonal tile and saves the intermediate transcript assignment result as a Parquet file.
+
+    """
+
+    from ..pre.boundary_tile import batch_transform_geometries
+    from ..pre.__init__ import _to_geometry
+
+    print("Reading transcripts file...")
+
+    ## only Xenium and Custom Tech supported for now
+
+    if gdf_transcripts is None:
+
+        if path_data is None:
+            raise ValueError("Either gdf_transcripts or path_data must be provided.")
+
+        trx = read_parquet_error_check(os.path.join(path_data, "transcripts.parquet"))
+
+        if os.path.isfile(os.path.join(path_data, "experiment.xenium")):
+            technology = "Xenium"
+
+            x = "x_location"
+            y = "y_location"
+            cell_id_col = "cell_id"
+            segmentation_approach = "default"
+
+            gdf_trx = gpd.GeoDataFrame(trx, geometry=gpd.points_from_xy(trx[x], trx[y]))
+
+        elif os.path.isfile(os.path.join(path_data, "segmentation_parameters.json")):
+
+            with open(os.path.join(path_data, "segmentation_parameters.json"), "r") as parameters_file:
+                parameters = json.load(parameters_file)
+                technology = parameters["technology"]
+
+            x = "x"
+            y = "y"
+            cell_id_col = "cell_index"
+            segmentation_approach = parameters["segmentation_approach"]
+
+            if type(trx['geometry'].iloc[0]) == str:
+                trx['geometry'] = trx['geometry'].apply(wkt.loads)
+
+            gdf_trx = gpd.GeoDataFrame(trx, geometry="geometry")
+            gdf_trx.set_geometry("geometry", inplace=True)
+
+        else:
+            raise ValueError("Unsupported or missing transcriptomics technology files.")
+
+    else:
+        gdf_trx = gdf_transcripts.copy()
+        gdf_trx.set_geometry("geometry", inplace=True)
+
+        technology = "PYTEST"
```

Comment:
lets try to have maybe an optional argument that says whether we're running a test rather than use technology